packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/HttpLogObfuscatorTests.kt

```diff
@@ -0,0 +1,354 @@
+        OBFUSCATED_FACEBOOK,
+        OBFUSCATED_GOOGLE_AUTH_CODE,
+        OBFUSCATED_JWT,
+        OBFUSCATED_CUSTOM_FUNCTION
+    }
+
+    @BeforeTest
+    fun setUp() {
+        channel = Channel(1)
+    }
+
+    private fun initApp(): TestApp {
+        return TestApp(
+            appName = "obfuscator",
+            logLevel = LogLevel.DEBUG,
+            customLogger = ObfuscatorLoggerInspector(channel),
+            initialSetup = { app, service ->
+                initializeDefault(app, service)
+                app.addFunction(TestAppInitializer.FIRST_ARG_FUNCTION)
+                app.addFunction(TestAppInitializer.SUM_FUNCTION)
+                app.addFunction(TestAppInitializer.NULL_FUNCTION)
+            }
+        )
+    }
+
+    @AfterTest
+    fun tearDown() {
+        channel.close()
+
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun nullObfuscator() = runBlocking {
+        app = TestApp(
+            appName = "null-obfuscator",
+            logLevel = LogLevel.DEBUG,
+            builder = { it.httpLogObfuscator(null) },
+            customLogger = NullObfuscatorLoggerInspector,
+            initialSetup = { app, service ->
+                initializeDefault(app, service)
+                app.addFunction(TestAppInitializer.FIRST_ARG_FUNCTION)
+                app.addFunction(TestAppInitializer.SUM_FUNCTION)
+                app.addFunction(TestAppInitializer.NULL_FUNCTION)
+            }
+        )
+
+        // Create user and log in
+        user = app.createUserAndLogin()
+
+        // Create API key
+        val key = user.apiKeyAuth.create("foo")
+        app.login(Credentials.apiKey(key.value!!))
+
+        // Login with Apple credentials fails as it normally does
+        assertFails {
+            app.login(Credentials.apple("apple"))
+        }
+
+        // Login with Facebook credentials fails as it normally does
+        assertFails {
+            app.login(Credentials.facebook("facebook"))
+        }
+
+        // Login with Google credentials fails as it normally does
+        assertFails {
+            app.login(Credentials.google("google-auth-code", GoogleAuthType.AUTH_CODE))
+        }
+        assertFails {
+            app.login(Credentials.google("google-id-token", GoogleAuthType.ID_TOKEN))
+        }
+
+        // Login with JWT fails as it normally does
+        assertFails {
+            app.login(Credentials.jwt("jwt"))
+        }
+
+        // Calling functions with arguments results in these not being obfuscated
+        with(user.functions) {
+            call<Double>(TestAppInitializer.FIRST_ARG_FUNCTION.name, 42.0)
+            call<Double>(TestAppInitializer.SUM_FUNCTION.name, 42.0, 1.0)
+            call<BsonNull>(TestAppInitializer.NULL_FUNCTION.name)
+        }
+
+        Unit
+    }
+
+    @Test
+    fun emailPassword_registerAndLogin() = runBlocking {
+        app = initApp()
+        createUserAndLoginAssertions()
```

Comment:
Again a bit backwards. What about just using the above mentioned `CustomLogCollector` and asserting that it contains the obfuscated entry and doesn't container a non-obfuscated entry. This would make the test way more explicit instead of having to compare the actual tests with the implementation of the log collector.